processing/index.ts

```diff
@@ -0,0 +1,62 @@
+import { $ } from 'bun'
+import { downloadFile, waitForFreshData } from './download'
+import { idFilter, tagFilter } from './filter'
+import { parseParameters } from './utils/parameters'
+import { processTopics } from './processTopics'
+import {
+  HASHES_DIR,
+  FILTER_DIR,
+  ID_FILTERED_FILE,
+  OSM_DOWNLOAD_DIR,
+  OSM_FILTER_DIR,
+} from './directories.const'
+import { TopicList } from './topics.const'
+import { directoryHasChanged, updateDirectoryHash } from './utils/hashing'
+import { writeMetadata } from './metadata'
+
+const params = parseParameters()
+
+$`mkdir -p ${OSM_DOWNLOAD_DIR} ${OSM_FILTER_DIR} ${HASHES_DIR}`
+
+if (params.waitForFreshData) {
+  await waitForFreshData(params.fileURL, 24, 10)
+}
+const { fileName, fileChanged } = await downloadFile(params.fileURL, params.skipDownload)
+
+if (fileChanged && !(await directoryHasChanged(FILTER_DIR))) {
+  await tagFilter(fileName)
+  updateDirectoryHash(FILTER_DIR)
+}
+
+if (params.idFilter && params.idFilter !== '') {
+  await idFilter(fileName, params.idFilter)
+  await processTopics(TopicList, ID_FILTERED_FILE, fileChanged, params)
+} else {
+  await processTopics(TopicList, fileName, fileChanged, params)
+}
+
+await writeMetadata(fileName, 0)
+
+// restart `tiles` container
+$`docker restart tiles`
+
+// call the frontend update hook
+try {
+  await fetch(`http://app:4000/api/private/post-processing-hook?apiKey=${params.apiKey}`)
+} catch {
+  console.warn(
+    'Calling the post processing hook failed. This is liekely due to the NextJS application not running.'
+  )
+}
+
+// call the cache warming hook
+if (params.warmCache) {
+  $`rm -rf /var/cache/nginx/*`
```

Comment:
Gibt es einen Weg das rm mit dem catch zu kombinieren? Vermutlich nicht … es sei denn, wir würden künstlich die API verlangsamen um 5 sec, was wiederum ja das error handling beeinflussen könnte(?)

Reply:
Verstehe nicht ganz was du meinst. Aber merke gerade, dass es eigentlich unabhänig von der warmCache Flag passieren sollte, da der cache in jedem Fall ungültig ist.